### ISSUE TITLE
[a11y]: Port ListBox fixes to v10 

### ISSUE THREAD

**Issue by alisonjoseph** [carbon-design-system/carbon#14069]
Bring the fixes for ListBox to v10 https://github.com/carbon-design-system/carbon/pull/13173

https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1687476988351489
![image](https://github.com/carbon-design-system/carbon/assets/2753488/6da9e43c-2e4f-47b7-8c21-718fbca75b3d)
